src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -120,7 +120,10 @@ export async function makeUtxoEngine(
         metadata.balance
       )
 
+      // Refresh the servers for the engine
       pluginState.addEngine(state)
+      await pluginState.refreshServers()
```

Comment:
This should not be here.
It should also not be in the resync method.

Well, kinda, it should be inside the pluginState.clearCache method like here: https://github.com/EdgeApp/edge-currency-bitcoin/blob/61b56f73e871f2fffccc55e800ad7cfabf24897d/src/plugin/pluginState.js#L140

Moving it here should not be the solution to the race condition you're having. 
I would like to go over this with you tomorrow to try and find a different solution.

Reply:
Let's discuss this point.